core/src/main/java/org/elasticsearch/action/admin/indices/analyze/AnalyzeRequest.java

```diff
@@ -82,35 +136,77 @@ public String analyzer() {
     }
 
     public AnalyzeRequest tokenizer(String tokenizer) {
-        this.tokenizer = tokenizer;
+        if (tokenizer != null) {
+            XContentType type = XContentFactory.xContentType(tokenizer);
```

Comment:
Trying to parse it as json or string is for handling request param's tokenizer and for AnalyzeRequestBuilder.
Should I add setTokenizer(Map<String, ?> tokenizer) to AnalyzeRequestBuilder and move parsing logic to outside AnalyzeRequest?